tests/integration/test_integration_legacy.py

```diff
@@ -21,9 +24,20 @@
 
 APP_URL = os.environ.get("APP_URL", 'http://localhost:5000')
 
-setup()
-# This implicitly tests the "/" path
-wait_for_service(APP_URL, "search2")
+
+def setup_module(module):
+    start_service(APP_URL)
+
+
+def teardown_module(module):
```

Comment:
It seems like starting and stopping the service after every step is kind of slow.   Couldn't it just start it if needed?

Reply:
I did consider this, but didn't think it worth taking more time to change that in this round, as I had already spent time moving the test setup from the startup script into the test api, because that process kept breaking and I'd have too manually manage the test containers. In practice, with just two tests, it doesn't really affect test run time very much.

That said, it would be an improvement and I'm happy to do it!

Reply:
Okay.  I thought there were more cycles involved.  Could it be as simple as just not doing the stop?  I'm fine either way though.